Add missing docstring sections in Special submodule

As part of my Google Season of Docs project with SymPy, I copy edited all of the docstrings in the Special submodule to reflect the new SymPy Docstring Style Guide. While editing, I came across some docstrings that were missing sections or had other issues that need to be addressed. Here is a list of all of the functions in the Special submodule that had an issue and what needs to be fixed:

1. Jn_zeros 
Parameters listed but does not have a parameters section

2. Bspline_basis, bspline_basis_set
In bspline_basis, parameters not pulling into HTML site
Both have parameters listed but do not have a parameters section

3. Interpolating_spline
Parameters listed but does not have a parameters section

4. Fdiff, eval (under DiracDelta)
Parameters listed but do not have a parameters section

5. Fdiff, eval (under Heaviside)
Parameters listed but do not have a parameters section

6. E1(z)
Missing examples section

7. Li (in error functions)
In the Examples section, the example for under the text “Defining the ``li`` function via an integral:” is missing

8. Hyper
Lists properties as part of this class that need to be their own docstrings with their own examples

9. Meijerg
Lists properties as part of this class that need to be their own docstrings with their own examples

10. Appellf1
Missing examples section

11. Jacobi_normalized
Parameters listed but does not have a parameters section

12. Ynm_c
Missing examples section
Parameters listed but does not have a parameters section

13. Znm
Missing examples section


CC @sympy/gsod

I am facing this error when I run ```make html``` inside doc:
```
rm -rf _build/logo
mkdir -p _build/logo
python ./generate_logos.py -d
INFO:root: File saved: ./_build/logo/sympy-notail.svg
INFO:root: File saved: ./_build/logo/sympy-notailtext.svg
INFO:root: File saved: ./_build/logo/sympy-notext.svg
DEBUG:root:command: rsvg-convert ./src/logo/sympy.svg -f png -o ./_build/logo/sympy-160px.png -h 160 -w 160
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./src/logo/sympy.svg -f png -o ./_build/logo/sympy-500px.png -h 500 -w 500
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notail.svg -f png -o ./_build/logo/sympy-notail-160px.png -h 160 -w 160
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notail.svg -f png -o ./_build/logo/sympy-notail-500px.png -h 500 -w 500
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notailtext.svg -f png -o ./_build/logo/sympy-notailtext-160px.png -h 160 -w 160
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notailtext.svg -f png -o ./_build/logo/sympy-notailtext-500px.png -h 500 -w 500
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notext.svg -f png -o ./_build/logo/sympy-notext-160px.png -h 160 -w 160
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notext.svg -f png -o ./_build/logo/sympy-notext-500px.png -h 500 -w 500
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./src/logo/sympy.svg -f png -o ./_build/logo/sympy-16px.png -h 16 -w 16
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./src/logo/sympy.svg -f png -o ./_build/logo/sympy-32px.png -h 32 -w 32
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./src/logo/sympy.svg -f png -o ./_build/logo/sympy-48px.png -h 48 -w 48
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./src/logo/sympy.svg -f png -o ./_build/logo/sympy-64px.png -h 64 -w 64
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notail.svg -f png -o ./_build/logo/sympy-notail-16px.png -h 16 -w 16
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notail.svg -f png -o ./_build/logo/sympy-notail-32px.png -h 32 -w 32
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notail.svg -f png -o ./_build/logo/sympy-notail-48px.png -h 48 -w 48
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notail.svg -f png -o ./_build/logo/sympy-notail-64px.png -h 64 -w 64
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notailtext.svg -f png -o ./_build/logo/sympy-notailtext-16px.png -h 16 -w 16
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notailtext.svg -f png -o ./_build/logo/sympy-notailtext-32px.png -h 32 -w 32
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notailtext.svg -f png -o ./_build/logo/sympy-notailtext-48px.png -h 48 -w 48
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notailtext.svg -f png -o ./_build/logo/sympy-notailtext-64px.png -h 64 -w 64
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notext.svg -f png -o ./_build/logo/sympy-notext-16px.png -h 16 -w 16
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notext.svg -f png -o ./_build/logo/sympy-notext-32px.png -h 32 -w 32
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notext.svg -f png -o ./_build/logo/sympy-notext-48px.png -h 48 -w 48
DEBUG:root:return code: 0
DEBUG:root:command: rsvg-convert ./_build/logo/sympy-notext.svg -f png -o ./_build/logo/sympy-notext-64px.png -h 64 -w 64
DEBUG:root:return code: 0
ERROR:root:Return code is not 0: Command: convert ./_build/logo/sympy-16px.png ./_build/logo/sympy-32px.png ./_build/logo/sympy-48px.png ./_build/logo/sympy-64px.png ./_build/logo/sympy-favicon.ico
ERROR:root:return code: 1
Makefile:143: recipe for target 'logo' failed
make: *** [logo] Error 1
```

Here are the commands I ran from the start to build:

```git clone git://github.com/sympy/sympy.git
cd sympy
git remote add github git@github.com:Soumi7/sympy.git
conda create -n sympy-dev-py35 python=3.5 mpmath flake8
source activate sympy-dev-py35
flake8 sympy
python setup.py install
git branch branch1
git checkout branch1
sudo apt-get install python-sphinx texlive-latex-recommended dvipng librsvg2-bin imagemagick docbook2x graphviz
sudo apt-get update
python -m pip install sphinx-math-dollar
ls
cd doc
make html
```

The logo generated but i got this error:

```
Logo generated.
mkdir -p src/.static
mkdir -p _build/html
mkdir -p _build/doctrees
mkdir -p src/modules
PYTHONPATH=..: sphinx-build -b html -d _build/doctrees -W --keep-going src _build/html
Running Sphinx v3.0.3

Extension error:
Could not import extension matplotlib.sphinxext.plot_directive (exception: No module named 'matplotlib')
Makefile:49: recipe for target 'html' failed
make: *** [html] Error 2
```
I installed matplotlib with this command:
```
sudo apt-get install python3-matplotlib
```
However, the **error still shows**. I had matplotlib installed on my system way before as well.

--- a/sympy/functions/special/bessel.py
+++ b/sympy/functions/special/bessel.py
@@ -984,6 +984,16 @@ def jn_zeros(n, k, method="sympy", dps=15):
 
     jn, yn, besselj, besselk, bessely
 
+    Parameters
+    ==========
+
+    n : integer
+        order of Bessel function
+
+    k : integer
+        number of zeros to return
+
+
     """
     from math import pi
 

--- a/sympy/functions/special/bsplines.py
+++ b/sympy/functions/special/bsplines.py
@@ -137,6 +137,20 @@ def bspline_basis(d, knots, n, x):
         >>> f = lambdify(x, b0)
         >>> y = f(0.5)
 
+    Parameters
+    ==========
+
+    d : integer
+        degree of bspline
+
+    knots : list of integer values
+        list of knots points of bspline
+
+    n : integer
+        $n$-th B-spline
+
+    x : symbol
+
     See Also
     ========
 
@@ -218,6 +232,17 @@ def bspline_basis_set(d, knots, x):
               (x**2/2 - 4*x + 8, (x >= 3) & (x <= 4)),
               (0, True))]
 
+    Parameters
+    ==========
+
+    d : integer
+        degree of bspline
+
+    knots : list of integers
+        list of knots points of bspline
+
+    x : symbol
+
     See Also
     ========
 
@@ -253,6 +278,20 @@ def interpolating_spline(d, x, X, Y):
     >>> interpolating_spline(3, x, [-2, 0, 1, 3, 4], [4, 2, 1, 1, 3])
     Piecewise((7*x**3/117 + 7*x**2/117 - 131*x/117 + 2, (x >= -2) & (x <= 1)),
             (10*x**3/117 - 2*x**2/117 - 122*x/117 + 77/39, (x >= 1) & (x <= 4)))
+
+    Parameters
+    ==========
+
+    d : integer
+        Degree of Bspline strictly greater than equal to one
+
+    x : symbol
+
+    X : list of strictly increasing integer values
+        list of X coordinates through which the spline passes
+
+    Y : list of strictly increasing integer values
+        list of Y coordinates through which the spline passes
 
     See Also
     ========

--- a/sympy/functions/special/delta_functions.py
+++ b/sympy/functions/special/delta_functions.py
@@ -132,6 +132,12 @@ class DiracDelta(Function):
         >>> diff(DiracDelta(x, 1)).fdiff()
         DiracDelta(x, 3)
 
+        Parameters
+        ==========
+
+        argindex : integer
+            degree of derivative
+
         """
         if argindex == 1:
             #I didn't know if there is a better way to handle default arguments
@@ -192,6 +198,14 @@ class DiracDelta(Function):
 
         >>> DiracDelta(x - 100).subs(x, 100)
         DiracDelta(0)
+
+        Parameters
+        ==========
+
+        k : integer
+            order of derivative
+
+        arg : argument passed to DiracDelta
 
         """
         k = sympify(k)
@@ -465,6 +479,12 @@ class Heaviside(Function):
         >>> diff(Heaviside(x)).fdiff()
         DiracDelta(x, 1)
 
+        Parameters
+        ==========
+
+        argindex : integer
+            order of derivative
+
         """
         if argindex == 1:
             # property number 1
@@ -527,6 +547,12 @@ class Heaviside(Function):
 
         >>> Heaviside(x - 100).subs(x, 105)
         1
+
+        Parameters
+        ==========
+
+        arg : argument passed by HeaviSide object
+        HO : boolean flag for HeaviSide Object is set to True
 
         """
         H0 = sympify(H0)

--- a/sympy/functions/special/error_functions.py
+++ b/sympy/functions/special/error_functions.py
@@ -1350,6 +1350,16 @@ def E1(z):
 
     This is equivalent to ``expint(1, z)``.
 
+    Examples
+    ========
+
+    >>> from sympy import E1
+    >>> E1(0)
+    expint(1, 0)
+
+    >>> E1(5)
+    expint(1, 5)
+
     See Also
     ========
 
@@ -1399,6 +1409,12 @@ class li(Function):
     1/log(z)
 
     Defining the ``li`` function via an integral:
+    >>> from sympy import integrate
+    >>> integrate(li(z))
+    z*li(z) - Ei(2*log(z))
+
+    >>> integrate(li(z),z)
+    z*li(z) - Ei(2*log(z))
 
 
     The logarithmic integral can also be defined in terms of ``Ei``:

--- a/sympy/functions/special/polynomials.py
+++ b/sympy/functions/special/polynomials.py
@@ -243,6 +243,17 @@ def jacobi_normalized(n, a, b, x):
     >>> jacobi_normalized(n, a, b, x)
     jacobi(n, a, b, x)/sqrt(2**(a + b + 1)*gamma(a + n + 1)*gamma(b + n + 1)/((a + b + 2*n + 1)*factorial(n)*gamma(a + b + n + 1)))
 
+    Parameters
+    ==========
+
+    n : integer degree of polynomial
+
+    a : alpha value
+
+    b : beta value
+
+    x : symbol
+
     See Also
     ========
 

--- a/sympy/functions/special/spherical_harmonics.py
+++ b/sympy/functions/special/spherical_harmonics.py
@@ -237,6 +237,26 @@ def Ynm_c(n, m, theta, phi):
     .. math::
         \overline{Y_n^m(\theta, \varphi)} := (-1)^m Y_n^{-m}(\theta, \varphi).
 
+    Examples
+    ========
+
+    >>> from sympy import Ynm_c, Symbol, simplify
+    >>> from sympy.abc import n,m
+    >>> theta = Symbol("theta")
+    >>> phi = Symbol("phi")
+    >>> Ynm_c(n, m, theta, phi)
+    (-1)**(2*m)*exp(-2*I*m*phi)*Ynm(n, m, theta, phi)
+    >>> Ynm_c(n, m, -theta, phi)
+    (-1)**(2*m)*exp(-2*I*m*phi)*Ynm(n, m, theta, phi)
+
+    For specific integers $n$ and $m$ we can evaluate the harmonics
+    to more useful expressions:
+
+    >>> simplify(Ynm_c(0, 0, theta, phi).expand(func=True))
+    1/(2*sqrt(pi))
+    >>> simplify(Ynm_c(1, -1, theta, phi).expand(func=True))
+    sqrt(6)*exp(I*(-phi + 2*conjugate(phi)))*sin(theta)/(4*sqrt(pi))
+
     See Also
     ========
 

